api/google_api/routes/Mailer.js

```diff
@@ -2,41 +2,70 @@ const express = require('express');
 const router = express.Router();
 const { SceGoogleApiHandler } = require('../util/SceGoogleApiHandler');
 const { verification } = require('../email_templates/verification');
+const { blastEmail } = require('../email_templates/blastEmail');
 const {
   OK,
   BAD_REQUEST
 } = require('../../util/constants').STATUS_CODES;
 const { googleApiKeys } = require('../../config/config.json');
 const { USER } = googleApiKeys;
 
+// Routing post /sendVerificationEmail calls the sendEmail function
+// and sends the verification email with the verification email template
 router.post('/sendVerificationEmail', async (req, res) => {
   const scopes = ['https://mail.google.com/'];
   const pathToToken = __dirname + '/../../config/token.json';
-  const apiHandler = new SceGoogleApiHandler(
-    scopes, pathToToken);
+  const apiHandler = new SceGoogleApiHandler(scopes, pathToToken);
   const tokenJson = await apiHandler.checkIfTokenFileExists();
 
   if (tokenJson) {
     if (apiHandler.checkIfTokenIsExpired(tokenJson)) {
       apiHandler.refreshToken();
     }
   } else {
-    apiHandler.getNewToken(false);
+    apiHandler.getNewToken();
   }
 
-  await verification(
-    USER, req.body.recipientEmail, req.body.recipientName
+  await verification(USER, req.body.recipientEmail, req.body.recipientName)
+    .then((template) => {
+      apiHandler
+        .sendEmail(template)
+        .then((_) => {
+          res.sendStatus(OK);
+        })
+        .catch((_) => {
```

Comment:
i think u can do a single `.catch` on the outside

Reply:
do you want me to remove the inner  `.catch` for the /sendBlastEmail router post as well?